src/TuxAnimations.js

```diff
@@ -45,75 +46,131 @@ var makeAnimation = function (transitions, customClassName) {
         }.bind(this));
       }.bind(this));
     },
-    componentWillEnter: function (callback) {
-      this.setAnimationDomNode('enter', callback);
-    },
 
-    componentWillLeave: function (callback) {
-      this.setAnimationDomNode('leave', callback);
-    },
-
-    render: function () {
+    componentWillMount: function () {
       //Change all custom props or add them if prop not defined in default
       for (var action in transitions) {
-        for (var css in transitions[action]) {
-
-          //Duration prop work-----------------------------------------
-          var duration = this.props.duration;
-          //Use default if no duration prop defined
-          if (duration) {
-            //Case to accept if duration prop is a number
-            if (typeof duration === "number") {
-              duration = duration + "ms";
+        if (action !== 'className') {
+          for (var css in transitions[action]) {
```

Comment:
Store `transitions[action]` in a variable so you don't need to look it up each time and so you can use it on line 64.